spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -383,6 +383,22 @@ def defined_private_method
         object.implemented
       end
 
+      it 'wraps proxies in a reentrant mutex so we dont trigger deadlocks' do
+        klass = Class.new { def message; end; }
+        called_for = []
+        RSpec.configuration.mock_with(:rspec) do |config|
+          config.before_verifying_doubles do |ref|
+            unless called_for.include? ref.target
+              called_for << ref.target
+              ::RSpec::Mocks.space.proxy_for(ref.target)
+            end
+          end
+          expect { allow(klass.new).to receive(:message) }.to_not raise_error
+          called_for.clear
```

Comment:
I don't see the connection with the ruby 2.2.0 warning.  The 2.2.0 warning I see is this:

```
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:404: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:439: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:444: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:449: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:457: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:461: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:469: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:475: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:484: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:511: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:515: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:516: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:521: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:532: warning: possible reference to past scope - klass
/Users/myron/code/rspec-dev/repos/rspec-mocks/spec/rspec/mocks/partial_double_spec.rb:546: warning: possible reference to past scope - klass
```

...because there's a `let(:klass)` below and a local variable `klass` in this spec.

I also tried commenting out `called_for.clear` and it didn't have any affect, so we can do away with it, I believe.